src/client/QXmppCallInviteManager.h

```diff
@@ -0,0 +1,123 @@
+// SPDX-FileCopyrightText: 2023 Tibor Csötönyi <work@taibsu.de>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#ifndef QXMPPCALLINVITEMANAGER_H
+#define QXMPPCALLINVITEMANAGER_H
+
+#include "QXmppClientExtension.h"
+#include "QXmppError.h"
+#include "QXmppJingleIq.h"
+#include "QXmppMessageHandler.h"
+#include "QXmppSendResult.h"
+#include "QXmppTask.h"
+
+class QXmppCallInviteManager;
+class QXmppCallInvitePrivate;
+class QXmppCallInviteManagerPrivate;
+
+class QXMPP_EXPORT QXmppCallInvite : public QObject
+{
+    Q_OBJECT
+public:
+    struct Rejected
+    {
+    };
+    struct Retracted
+    {
+    };
+    struct Left
+    {
+    };
+
+    using Result = std::variant<Rejected, Retracted, Left, QXmppError>;
+
+    QXmppCallInvite(QXmppCallInviteManager *manager);
+    ~QXmppCallInvite();
+
+    QXmppTask<QXmpp::SendResult> invite(
+        bool audio = true,
+        bool video = false,
+        std::optional<QXmppCallInviteElement::Jingle> jingle = std::nullopt,
+        std::optional<QVector<QXmppCallInviteElement::External>> external = std::nullopt);
```

Comment:
Maybe this should be private and only be accessed by the manager.

Reply:
My thought here was that the call invites are only created by the manager and there's no API to create the call "manually". Was your plan that the users of QXmpp can "manually" create a `CallInvite()` with a manager and then calling `invite()` on the call invite?

If (only) `CallInviteManager::invite()` should be used to create invites, this and the `CallInvite(CallInviteManager *)` ctor should probably be made private (with friend class for the manager).